### ISSUE TITLE
Question about the smoothing method applied over Figure 4.

### ISSUE THREAD

**Issue by thu-yao-01-luo** [huggingface/datablations#12]
When I check the [plotting script of Figure 4](https://colab.research.google.com/drive/1i00FUdVp0Oj-Qw40ITLSGXwzuxpqBjmw?usp=sharing#scrollTo=abWdyOLYMKhy) in your repo, I notice that the smoothing methods are quite different for training loss and validation loss. Besides, with regard to the validation loss, I find that you first apply the **savgol_filter** and then the **exponential moving average**. My question are as follows.
1. Are there any motivations for the choices of the smoothing methods, or they are just the results of careful tuning?
2. Why do you choose different smoothing methods for training and validation losses? I mean, if the smoothing method with  **savgol_filter** seems better for validation loss, why do you apply it over the training loss curve? Besides, the weight of EMA of training loss curve is 0.999, I worry about the over-smoothing issue since the trend would be quite different from the original one.

Appendix(the code snippet):
```
def get_data_csv(model, model_id, smooth_data=False):
    if LOSS == "training":
        if not(os.path.exists(f"{model}_{LOSS}.csv")):
            !wget https://huggingface.co/datasets/datablations/scripts/raw/main/tb{LOSS}/{model}_{LOSS}.csv
    else:
        if not(os.path.exists(f"{model}.csv")):
            !wget https://huggingface.co/datasets/datablations/scripts/raw/main/tb{LOSS}/{model}.csv
    path = f"{model}.csv" if LOSS == 'validation' else f"{model}_{LOSS}.csv"
    df = pd.read_csv(path, index_col=0)
    if LOSS == "training":
        df['value'] = smooth(df["value"].values.tolist(), weight=0.999)
    elif smooth_data:
        df[["value"]] = df[["value"]].apply(savgol_filter,  window_length=3, polyorder=2)
        df['value'] = smooth(df["value"].values.tolist(), weight=0.85)
    return df.step.values, df.value.values
```

**Comment by Muennighoff** [comment]
1. The motivations are using a common & well-proven method (i.e. exponential) & making it visually easy to digest
2. Validation losses are not smoothed except for the 8.7B parameter runs. The colab should be runnable, so if you remove the smoothing for 8.7B, the validation plot would look like below:
 
![image](https://github.com/huggingface/datablations/assets/62820084/79a7966e-4f40-497a-8921-b5708b178021)

As you can see for the 8.7B the pattern is overall the same but there are a ton of spikes. The savgol filter actually does not make a difference at all, so I removed it from the code, so it is just exponential moving average smoothing with a weight of 0.85 for the 8.7B validation runs.

For training runs smoothing is indeed always used with 0.999 mainly to reduce spikes etc, below is how it would like with 0.9.  I think the trend is the same and smoothing as used in the paper helps making the trend easier to see.


![image](https://github.com/huggingface/datablations/assets/62820084/9f9eef01-4e77-4267-a54b-023e245753d9)



I have updated the caption of Figure 4 with the final sentence in the below pic & will update the paper on arxiv in the coming months, hope this helps!


<img width="454" alt="Screenshot 2024-03-22 at 11 19 41 AM" src="https://github.com/huggingface/datablations/assets/62820084/2e525f50-f3a3-417d-81b9-b89e7f1be7c1">
